jwst/datamodels/schemas/darkMIRI.schema.yaml

```diff
@@ -0,0 +1,27 @@
+title: MIRI Dark current data model
+allOf:
+- $ref: core.schema.yaml
+- $ref: referencefile.schema.yaml
+- type: object
+  properties:
+    data:
+      title: MIRI Dark current array
+      fits_hdu: SCI
+      default: 0.0
+      ndim: 4
+      datatype: float32
+    dq:
+      title: 4-D data quality array for all planes
+      fits_hdu: DQ
+      default: 0
+      ndim: 4
+      datatype: uint32
```

Comment:
I just copied the dark.schema.yaml but modified it to be 4D.
The values are floats and not unit8. So I think they need to
be int32.
The MIRI darks are not really darks but a relative correction
from time = 0. If we need to we could revise how the MIRI darks are
created and look at rounding up or truncating the values and
they delivering those integer values in the reference file instead.

Jane

On 07/21/2016 06:39 PM, Howard Bushouse wrote:
>
> In jwst/datamodels/schemas/darkMIRI.schema.yaml 
> <https://github.com/STScI-JWST/jwst/pull/118#discussion_r71816097>:
>
> > +- $ref: core.schema.yaml
> > +- $ref: referencefile.schema.yaml
> > +- type: object
> > +  properties:
> > +    data:
> > +      title: MIRI Dark current array
> > +      fits_hdu: SCI
> > +      default: 0.0
> > +      ndim: 4
> > +      datatype: float32
> > +    dq:
> > +      title: 4-D data quality array for all planes
> > +      fits_hdu: DQ
> > +      default: 0
> > +      ndim: 4
> > +      datatype: uint32
>
> Does this need to be as big as int32? If only a few DQ bits are used 
> in the reference data, this could be reduced to something like uint8. 
> That still gives you 8 separate DQ bits to work with and saves a lot 
> of space.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/STScI-JWST/jwst/pull/118/files/76105243e0379a5934adbb1034a1377a5d1cec05#r71816097>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/ARonWbtFc-AGfkonYVMsuA54MCLH2Fvgks5qYB9TgaJpZM4JSExt>.
>